### PR TITLE
Snapshot: add some kubernetes deployments and make container target

### PR DESCRIPTION
**Pull request by tsmetana**
This should help using snapshot controller: Makefile in snapshot directory can now create containers and I have added also some deployments for kubernets (HostPath and AWS).

I have bundled a small fix for snapshotter: we ignored error from API server when GETting the VolumeSnapshot. That lead to somewhat mysterious errors.

PTAL.

### CODE REVIEW

File: snapshot/deploy/kubernetes/aws/snapshot-rbac.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    resources: ["persistentvolumeclaims"]
+    verbs: ["get", "list", "watch", "update"]
+  - apiGroups: ["storage.k8s.io"]
+    resources: ["storageclasses"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["list", "watch", "create", "update", "patch"]
+  - apiGroups: [""]
+    resources: ["services", "endpoints"]
```
Comment (rootfs) [inline snapshot/deploy/kubernetes/aws/snapshot-rbac.yaml]: why do we need services and endpoints?

Comment (tsmetana) [reply snapshot/deploy/kubernetes/aws/snapshot-rbac.yaml]: We don't. I just started with nfs as template and didn't realize this part should go away.

Comment (rootfs) [reply snapshot/deploy/kubernetes/aws/snapshot-rbac.yaml]: i don't know if nfs deploy needs services/endpoints @wongma7

**Comment by wongma7** [comment]
Yes, only the nfs needs these

File: snapshot/Makefile [hunk 2]
```diff
@@ -12,13 +12,32 @@
...
 	go build -o _output/bin/snapshot-provisioner cmd/snapshot-pv-provisioner/snapshot-pv-provisioner.go
 
 clean:
+	-rm -f deploy/snapshot-controller
+	-rm -f deploy/snapshot-provisioner
 	-rm -rf _output
+
+container: controller quick-container
+
+container-quick:
```
Comment (jsafrane) [inline snapshot/Makefile]: it's `quick-container` on the previous line

Comment (tsmetana) [reply snapshot/Makefile]: Oops. I could not decide how to name the target... And this is the outcome. Will fix it.

**Comment by tsmetana** [comment]
So... any chance of merging this one? Is there still something I should change?

**Pull request merged**